main/provider/index.ts

```diff
@@ -576,7 +574,11 @@ export class Provider extends EventEmitter {
       params: orderedParams
     }
 
-    let [from = '', typedData = {}, ...additionalParams] = payload.params
+    let [from = '', typedData, ...additionalParams] = payload.params as [string, LegacyTypedData | TypedData | string, ...unknown[]]
```

Comment:
we can add these types to a request and make this method take a typed payload. check out `@types/frame/rpc.d.ts` for some examples